.github/workflows/publish-canary.yml

```diff
@@ -0,0 +1,38 @@
+name: Publish @canary to NPM
+
+permissions:
+  contents: write
+  pull-requests: write
+  actions: read
+  statuses: write
+
+on:
+  # So that we publish for every push to `main`, despite tests
+  push:
+    branches: [main]
+  workflow_dispatch:
```

Comment:
Take note – any merge to `main` is a new `@canary` for testing something without needing acceptance tests 1st.